### ISSUE TITLE
Blog: Write a blog post on no-unsafe-* rules

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6415]
## Overview

One of the biggest user requests I see in the wild for typescript-eslint is banning `any`s from sneaking into the codebase. We have a suite of rules: 

* [`@typescript-eslint/no-unsafe-argument`](https://typescript-eslint.io/rules/no-unsafe-argument)
* [`@typescript-eslint/no-unsafe-assignment`](https://typescript-eslint.io/rules/no-unsafe-assignment)
* [`@typescript-eslint/no-unsafe-call`](https://typescript-eslint.io/rules/no-unsafe-call)
* [`@typescript-eslint/no-unsafe-member-access`](https://typescript-eslint.io/rules/no-unsafe-member-access)
* [`@typescript-eslint/no-unsafe-return`](https://typescript-eslint.io/rules/no-unsafe-return)

I'd like to write a blog post explaining:

1. Why you'd want to ban `any` even with `noImplicitAny` enabled
2. Why `unknown` is a better alternative
3. How each of those rules helps

**Comment by Zamiell** [comment]
I think the blog should first and foremost tell users to use:

```ts
extends: [
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
    "plugin:@typescript-eslint/strict",
]
```

Which enables the rules mentioned in the OP automatically. Doing this `extends` should be considered best practice in the TypeScript ecosystem.

I think that once we have given users the TLDR of what to do, then we can go into the gory details of what all the individual rules actually do.

**Comment by mmkal** [comment]
In the absence of a blog post, does anyone know why there are 5 `no-unsafe-*` rules (as well as `no-explicit-any`), rather than a single `@typescript-eslint/no-any` rule?

The way it is can make it very cumbersome to disable blocks of code that uses any (for example, code that's being migrated from a project that didn't have these rules).

I was thinking about opening an issue along these lines but thought this could be a good place to ask first.

**Comment by JoshuaKGoldberg** [comment]
Oop, sorry @mmkal - I meant to respond to your question in April but lost track of it. It's a good one and tentatively I'd be in favor of adding a docs FAQ point about why groups of rules like `no-unsafe-*` are split up. That'd be a good separate issue as you suggested.

In general, I've always interpreted it as twofold:

* Conceptually: to keep to "one area of report, one lint rule"
* Practically: to make it easier to configure them granularly

By splitting them up, users can granularly reason about and configure them. While onboarding a codebase to the rules, you can more easily pick-and-choose which ones are disabled, warnings, or errors.

**Comment by bradzacher** [comment]
As the guy that implemented them josh is partially correct.
The main reason they are separate rules is because it was so much simpler to implement them that way.
The rules themselves comprise of ~1300 lines of code and a single rule that is that big is so so so hard to wrap your head around and understand.

I started with one rule to rule them all but as soon as I went to implement a second set of checks I realised that the new logic mostly separate to the previous logic - so rather than intermingling them I just split them up.
This also made it easy to land and release self-contained, completed rules to people rather than landing one massive rule at once, or landing a small chunk of one rule and adding more to that one rule over time (which would have been a breaking change).

TL;DR -- it's ultimately just easier for us to maintain!

**Comment by FloEdelmann** [comment]
Quick heads up: The [newly released](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v8.15.0) [`@typescript-eslint/no-unsafe-type-assertion`](https://typescript-eslint.io/rules/no-unsafe-type-assertion/) rule should also be covered in the blog post.

**Comment by JoshuaKGoldberg** [comment]
#10362 is in review.

> Quick heads up: The [newly released](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v8.15.0) [`@typescript-eslint/no-unsafe-type-assertion`](https://typescript-eslint.io/rules/no-unsafe-type-assertion/) rule should also be covered in the blog post.

Tempting, but the post is kind of long as-is and the rule doesn't actually cover `any` specifically. I think we could have a separate post for assertion best practices.
